stackrox/stackrox
# ROX-14947: Plug UI NetworkGraph page (2.0) on new API for clusters and namespaces in scope

## Description

Currently, the NetworkGraph pages in the UI requires `Cluster`, `Namespace`, `NetworkGraph` and `Deployment` permissions to display.

The `Cluster` permission is problematic as it could expose cluster configuration information that is not relevant to activities around network graphs.
A new API has been created to expose only limited cluster and namespace information for the clusters and namespaces visible by the user access scopes.

This change aims at integrating this new API in the UI NetworkGraph pages.

In its initial state, the `Network Graph (2.0)` page is slightly more complex than the `1.0` version. The Namespace drop-down uses richer information than just namespace names and IDs: for each namespace, the count of deployments in the namespace is shown.

![Screenshot 2023-03-14 at 12 21 42](https://user-images.githubusercontent.com/91869377/224986063-1a99ea7a-8c5b-40d7-b6f6-cc0ad5c387cf.png)

This PR focuses on plugging the new API showing restricted cluster and namespace information to fill the navigation drop downs. ~~The GraphQL query that used to provide the data for the Namespace drop-down is replaced by a first call to list the namespaces, then by a call to the deployment API to fetch the deployment count for each namespace.~~

~~Currently, the approach has a bad side-effect: although the namespace list and deployment counts retrieval occur quickly after the cluster selection, the namespace drop down can still be shown empty for 15 to 30 seconds. It looks like the hook that enriches the namespace information with deployment count does not trigger re-rendering of the `NetworkBradcrumbs` component of the NetworkGraph page.~~

After discussion, the decision was made to drop the support for the deployment count badge in the drop-down.

## Checklist
~~- [ ] Unit test and regression tests added~~
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

Mainly CI testing should ensure stability of the pre-existing features.

Manual testing:
- Deploy Central with the new scope info API
- Ensure web developer tools are open in the client browser
- Open the `Network Graph (2.0)` page 
- Check the proper API is called for cluster drop-down population
- Select a cluster
- Check the proper API is called for namespace drop-down population
- Open the namespace drop-down
- Check the namespace list and the absence of badge
